pkg/validation/internal/multiarch.go

```diff
@@ -0,0 +1,513 @@
+
+// MultiArchValidator store the data to perform the tests
+type MultiArchValidator struct {
+	// InfraArchLabels store the arch labels (i.e amd64, ppc64le) from
+	// operatorframework.io/arch.<GOARCH>: supported
+	InfraArchLabels []string
+	// InfraSOLabel store the OS labels from
+	// operatorframework.io/so.<GOARCH>: supported
+	InfraSOLabels []string
+	// Images stores the images defined in the bundle with the platform.arch supported
+	Images map[string][]Platform
+	// AllArchFound contains a map of the arch types found
+	AllArchFound map[string]string
+	// AllOsFound contains a map of the so(s) found
+	AllOsFound map[string]string
+	// Store the bundle load
+	Bundle *manifests.Bundle
+	// ContainerTool defines the container tool which will be used to inspect the images
+	ContainerTool string
+	// Warns stores the errors faced by the validator to return the warnings
+	Warns []error
+}
+
+// ManifestInspect store the data obtained by running container-tool manifest inspect <IMAGE>
+type ManifestInspect struct {
+	ManifestData []ManifestData `json:"manifests"`
+}
+
+// ManifestData store the platforms
+type ManifestData struct {
+	Platform Platform `json:"platform"`
+}
+
+// Platform store the Architecture and OS supported by the image
+type Platform struct {
+	Architecture string `json:"architecture"`
+	OS           string `json:"os"`
+}
+
+func multipleArchitecturesValidate(objs ...interface{}) (results []errors.ManifestResult) {
+	// Obtain the k8s version if informed via the objects an optional
+	var containerTool = ""
+	for _, obj := range objs {
+		switch obj.(type) {
+		case map[string]string:
+			containerTool = obj.(map[string]string)[ContainerToolsKey]
+			if len(containerTool) > 0 {
+				break
+			}
+		}
+	}
+
+	for _, obj := range objs {
+		switch v := obj.(type) {
+		case *manifests.Bundle:
+			results = append(results, validateMultiArchWith(v, containerTool))
+		}
+	}
+	return results
+}
+
+func validateMultiArchWith(bundle *manifests.Bundle, containerTool string) errors.ManifestResult {
+	result := errors.ManifestResult{}
+	if bundle == nil {
+		result.Add(errors.ErrInvalidBundle("Bundle is nil", nil))
+		return result
+	}
+
+	result.Name = bundle.Name
+
+	if bundle.CSV == nil {
+		result.Add(errors.ErrInvalidBundle("Bundle csv is nil", bundle.Name))
+		return result
+	}
+
+	containerTool, err := validateContainerTool(containerTool)
+	if err != nil {
+		result.Add(errors.ErrFailedValidation(err.Error(), bundle.CSV.GetName()))
+		return result
+	}
+
+	multiArchValidator := MultiArchValidator{Bundle: bundle, ContainerTool: containerTool}
+	multiArchValidator.validate()
+
+	for _, err := range multiArchValidator.Warns {
+		result.Add(errors.WarnFailedValidation(err.Error(), bundle.CSV.GetName()))
+	}
+
+	return result
+}
+
+// validateContainerTool verifies if the container tool informed is valid
+func validateContainerTool(containerTool string) (string, error) {
+	if len(containerTool) == 0 || containerTool == "none" {
```

Comment:
```suggestion
	if len(containerTool) == 0 || strings.ToLower(containerTool) == "none" {
```

Reply:
Good point. 
Changed on top for we ensure that we will have the lower case always.

Reply:
See:

```
// Check the key values informed
			containerTool = obj.(map[string]string)[ContainerToolsKey]
			if len(containerTool) > 0 {
				// Make lower for we compare and use it
				containerTool = strings.ToLower(containerTool)
				break
			}
```
We ensure that the string is informed as the value of the key will be the lower case always so that here we just check if the value is or is not valid. Is that make sense?